next_sparseconvnet/utils/train_utils.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+import torch
+import sys
+import sparseconvnet as scn
+from .data_loaders import DataGen, collatefn, LabelType
+from next_sparseconvnet.networks.architectures import UNet
+
+def IoU(true, pred, nclass = 3):
+    """
+        Intersection over union is a metric for semantic segmentation.
+        It returns a IoU value for each class of our input tensors/arrays.
+    """
+    eps = sys.float_info.epsilon
+    confusion_matrix = np.zeros((nclass, nclass))
+
+    for i in range(len(true)):
+        confusion_matrix[true[i]][pred[i]] += 1
+
+    IoU = []
+    for i in range(nclass):
+        IoU.append((confusion_matrix[i, i] + eps) / (sum(confusion_matrix[:, i]) + sum(confusion_matrix[i, :]) - confusion_matrix[i, i] + eps))
+    return IoU
+
+
+def train_one_epoch(epoch_id, net, criterion, optimizer, loader):
```

Comment:
Since you are explicitly calling iuo metrics inside you should rename this to `train_one_epoch_segmentation`or so. Eventually we can merge classification and segmentation in one function, but lets leave that for later